daemons/controld/controld_messages.c

```diff
@@ -909,8 +909,12 @@ handle_node_info_request(const xmlNode *msg)
                            pcmk_is_set(controld_globals.flags,
                                        controld_has_quorum));
 
-    // Check whether client requested node info by ID and/or name
-    crm_element_value_int(msg, PCMK_XA_ID, &node_id);
+    /* Check whether client requested node info by ID and/or name
+     *
+     * @TODO A Corosync-layer node ID is of type uint32_t. We should be able to
+     * handle legitimate node IDs greater than INT_MAX, but currently we do not.
```

Comment:
Just use crm_element_value_ll() and truncate to uint32_t

Reply:
yeah I can do that, I was just tired and figured there are probably more places where we can overflow like this

Reply:
I'd rather not deal with this right now, as we're nearing EOD/EOW and fixing it properly requires several changes elsewhere, including changing a data type in `pcmk_controld_api_reply_t`.

Reply:
We know it's uint32_t, so just use a local long long variable for the parsing and set node_id to it

Reply:
yes it's easy to do here, but there are other places it ought to be done and that's not likely to happen before EOD, I'm about to go to lunch

It's not necessary for this PR, it would just fix huge node IDs, which have likely always been broken